File: Reactivities/reactivities.ui/src/app/models/activity.ts
import { Profile } from "./profile";

export interface Activity {
    title: string;
    category: string;
    description: string;
    city: string;
    venue: string;
    date: Date;
    id: string;
    hostUsername: string;
    isCancelled: boolean;
    attendees: Profile[];
    isGoing: boolean;
    isHost: boolean;
    host?: Profile;
}

export class Activity implements Activity {
    constructor(init?: ActivityFormValues){
        Object.assign(this, init);
    }
}

export class ActivityFormValues {
   id?: string = undefined;
   title: string = '';
   category: string = '';
   description: string = '';
   date: Date | null = null;
   venue: string = '';
   city: string = '';

   constructor(activity?: ActivityFormValues){
        if(activity){
            this.id = activity.id;
            this.title = activity.title;
            this.category = activity.category;
            this.description = activity.description;
            this.date = activity.date;
            this.venue = activity.venue;
            this.city = activity.city;
        }
   }
}

File: Reactivities/Reactivities.Core/Models/Activity.cs
using Reactivities.Core.Models.Base;
using Reactivities.Core.Models.Identity;
using Reactivities.Core.Models.Relations;

namespace Reactivities.Core.Models;

public class Activity : BaseEntity
{
    public string Title { get; set; }   
    
    public string Category { get; set; }
    
    public string Description { get; set; }

    public string City { get; set; }

    public string Venue { get; set; }   
    
    public bool IsCancelled { get; set; }
    
    public DateTimeOffset Date { get; set; }

    public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>();

    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
}